cardInfoAPI-Level1.yaml

```diff
@@ -0,0 +1,1402 @@
+  - name: Card-tokens
+    description: Operations for retrieving information about tokens associated with payment cards.
+  - name: Transactions
+    description: Operations for retrieving information about transactions and authorizations done with payment cards or tokens.
+
+security:
+  - ApiKeyAuth: []
+  - OAuth2:
+      - read
+      - write
+
+paths:
+  /cards:
+    get:
+      summary: Provides a list of cards which match the provided query parameters.
+      description: >
+        This operation enables the retrieval of a list of payment cards. Payment cards can be physical or virtual cards.
+        In order to prevent API consumers from having to fulfill PCI DSS requirements, no full PAN or other data relevant for this regulation are included.
+        The API consumer is expected to have a non-card related identifier for the customer for whom data should be returned (e.g. card account number, bank account number).
+      tags:
+        - Cards
+      parameters:
+        - in: query
+          name: contract_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/ContractReference'
+          description: Retrieves all cards related to a contract. Value has to be a functional reference to the contract that can be known by the end customer.
+        - in: query
+          name: person_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/PersonReference'
+          description: Retrieves all cards related to an individual person/cardholder. Value has to be a functional reference to the person that can be known by the end customer.
+        - in: query
+          name: card_status
+          required: false
+          schema:
+            $ref: '#/components/schemas/CardStatus'
+          description: Retrieves all cards in a certain status. Only one status can be filtered at a time.
+        - in: query
+          name: card_account_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/CardAccountReference'
+          description: Retrieves all cards related to a card account (not bank account). Value has to be a functional reference to the account that can be known by the end customer.
+        - in: query
+          name: bank_account_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/BankAccountReference'
+          description: Retrieves all cards related to a bank account (not card account). Value has to be a functional reference to the account that can be known by the end customer and is usually an IBAN.
+        - $ref: '#/components/parameters/limit'
+        - $ref: '#/components/parameters/offset'
+        - $ref: '#/components/parameters/client_in_header'
+        - $ref: '#/components/parameters/correlation_in_header'
+        - $ref: '#/components/parameters/agent_in_header'
+      responses:
+        '200':
```

Comment:
Please first establish which values are relevant + optional and where, where applicable, structured values are expected.
- CardID: Does this field have a uniform length throughout Switzerland? Currently, each PSP has its own ideas.
- PAN4 digits: Necessary? For a better structure, the first BIN letters and the last 4 digits could help

Reply:
The mandatory and optional fields are defined in the Swagger.
Card Id will be a standardized uuid.
PAN4 digits: only the last four digits will be provided. We want to avoid that API consumers build implicit logic based on the BIN. Separate attributes are available to identify product types.